src/BinSkim.Driver/RuleTelemetryLogger.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.CodeAnalysis.Sarif;
+
+namespace Microsoft.CodeAnalysis.IL
+{
+    /// <summary>
+    /// A logger that records rule summary telemetry by aggregating result kinds for each rule.
+    /// </summary>
+    internal sealed class RuleTelemetryLogger : IAnalysisLogger
+    {
+        // Application Insights event names
+        internal const string RuleSummaryEventName = "RuleSummary";
+        internal const string AnalysisStoppedEventName = "AnalysisStopped";
+
+        // Application Insights property names
+        internal const string RuleIdPropertyName = "RuleId";
+        internal const string RuntimeConditionsPropertyName = "Flags";
```

Comment:
>Flags [](http://example.com/codeflow?start=63&length=5)

Is this the right name here? Will keep an eye open for references. These runtime conditions specify various things we observed at runtime, like exceptions, whether an error was produced. Did you prefer 'Flags' over something like 'RuntimeConditions' from a general spirit of keeping terms more concise?

Reply:
Yes, I was trying to be terse. Easy enough to change, of course.

Reply:
In the latest iteration, I'm using a RequestTelemetry with a name of "Analysis"